katello/katello-backup

```diff
@@ -59,70 +125,77 @@ def create_directories(directory)
+    run_cmd("runuser - postgres -c 'pg_dump -Fc candlepin > #{@dir}/candlepin.dump'")
+    puts "Done."
+  when 'mongodb'
+    puts "Backing up mongo online schema... "
+    run_cmd('mongodump --host localhost --out mongo_dump')
+    puts "Done."
+  when 'pulp'
+    puts "Backing up Pulp data... "
+    matching = false
+   
+    until matching
+      checksum1 = `find /var/lib/pulp -printf '%T@\n' | md5sum`
+      create_pulp_data_tar
+      checksum2 = `find /var/lib/pulp -printf '%T@\n' | md5sum`
+      matching = (checksum1 == checksum2)
+    end
+    puts "Done."
+  end
 end
 
-def backup_pulp_online
-  puts "Backing up Pulp data... "
-  matching = false
- 
-  until matching
-    checksum1 = `find /var/lib/pulp -printf '%T@\n' | md5sum`
+def offline_backup(database)
+  case database
+  when 'postgres'
+    puts "Backing up postgres db... "
+    run_cmd('tar --selinux --create --file=pgsql_data.tar --listed-incremental=.postgres.snar /var/lib/pgsql/data/')
+    puts "Done."
+  when 'mongodb'
+    puts "Backing up mongo db... "
+    run_cmd('tar --selinux --create --file=mongo_data.tar --listed-incremental=.mongo.snar --exclude=mongod.lock /var/lib/mongodb/')
+    puts "Done."
+  when 'pulp'
+    puts "Backing up Pulp data... "
     create_pulp_data_tar
-    checksum2 = `find /var/lib/pulp -printf '%T@\n' | md5sum`
-    matching = (checksum1 == checksum2)
+    puts "Done."
   end
-  puts "Done."
-end
-
-def backup_db_offline
-  puts "Backing up postgres db... "
-  run_cmd('tar --selinux --create --file=pgsql_data.tar --listed-incremental=.postgres.snar /var/lib/pgsql/data/')
-  puts "Done."
-
-  puts "Backing up mongo db... "
-  run_cmd('tar --selinux --create --file=mongo_data.tar --listed-incremental=.mongo.snar --exclude=mongod.lock /var/lib/mongodb/')
-  puts "Done."
-end
-
-def backup_pulp_offline
-  puts "Backing up Pulp data... "
-  create_pulp_data_tar
-  puts "Done."
 end
 
 def create_pulp_data_tar
   run_cmd('tar --selinux --create --file=pulp_data.tar --exclude=/var/lib/pulp/katello-export --listed-incremental=.pulp.snar /var/lib/pulp/ /var/www/pub/')
 end
 
 def compress_files
-  psql = spawn('gzip', 'pgsql_data.tar')
+  psql = spawn('gzip', 'pgsql_data.tar') if @databases.include? "postgres"
   mongo = spawn('gzip', 'mongo_data.tar')
-  Process.wait(psql)
+  Process.wait(psql) if @databases.include? "postgres"
   Process.wait(mongo)
 end
 
-def generate_metadata
-  os_version = `cat /etc/redhat-release`.chomp
-  plugin_list = `foreman-rake plugin:list | grep 'Foreman plugin: '`.lines
+def plugin_list
   plugins = []
+  plugin_list = `foreman-rake plugin:list | grep 'Foreman plugin: '`.lines
   plugin_list.each do |line|
     plugin = line.split
     plugins << "#{plugin[2].chop}-#{plugin[3].chop}"
   end
+  plugins
+end
+
+def generate_metadata
+  os_version = `cat /etc/redhat-release`.chomp
+  plugins = @is_foreman_proxy_content ? "none" : plugin_list
```

Comment:
We should include a list of smart proxy plugins if on the main server and on an external proxy. This will help identify what features are present.